x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -514,6 +522,12 @@ export function App({
     }
   };
 
+  const lastKnownDocRef = useRef(state.lastKnownDoc);
```

Comment:
Those are save to pass in via refs because they are only used in the `onChange` callback to compare